compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -396,34 +398,53 @@ impl<'a, 'tcx> AbstractConstBuilder<'a, 'tcx> {
             | ExprKind::Field { .. }
             | ExprKind::ConstBlock { .. }
             | ExprKind::Adt(_) => self.error(
+                Some(node.span),
+                "unsupported operation in generic constant, this may be supported in the future",
```

Comment:
I think we should split this arm up a bit to give more specific error messages. I can imagine someone might be confused as to what the unsupported thing is when doing `foo.bar` or building a tuple or making an adt etc